addons/mail/models/mail_message.py

```diff
@@ -826,6 +826,10 @@ def _message_format(self, fnames):
             else:
                 author = (0, message_sudo.email_from)
 
+            # Sender
+            sender = (message_sudo.create_uid.partner_id.id, message_sudo.create_uid.partner_id.display_name)
+            author_has_user = len(message_sudo.author_id.user_ids) > 0
```

Comment:
This needs to be adapted to take guest into account (if `create_uid` is public user in particular).